kola/tests/kubeadm/templates.go

```diff
@@ -0,0 +1,384 @@
+// Copyright 2021 Kinvolk GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package kubeadm
+
+var (
+	workerConfig = `systemd:
+  units:
+    - name: prepare-cni-plugins.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=Unpack CNI plugins to /opt/cni/bin
+        ConditionPathExists=!/opt/cni/bin
+        [Service]
+        Type=oneshot
+        RemainAfterExit=true
+        Restart=on-failure
+        Environment=CNI_VERSION={{ .CNIVersion }}
+        ExecStartPre=/usr/bin/mkdir --parents /opt/cni/bin
+        ExecStartPre=/usr/bin/tar -v --extract --file "/opt/cni-plugins-linux-amd64-${CNI_VERSION}.tgz" --directory /opt/cni/bin --no-same-owner
+        ExecStart=/usr/bin/rm "/opt/cni-plugins-linux-amd64-${CNI_VERSION}.tgz"
+        [Install]
+        WantedBy=multi-user.target
+    - name: prepare-critools.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=Unpack CRI tools to /opt/bin
+        ConditionPathExists=!/opt/bin/crictl
+        [Service]
+        Type=oneshot
+        RemainAfterExit=true
+        Restart=on-failure
+        Environment=CRICTL_VERSION={{ .CRIctlVersion }}
+        Environment=DOWNLOAD_DIR={{ .DownloadDir}}
+        ExecStartPre=/usr/bin/mkdir --parents "${DOWNLOAD_DIR}"
+        ExecStartPre=/usr/bin/tar -v --extract --file "/opt/crictl-${CRICTL_VERSION}-linux-amd64.tar.gz" --directory "${DOWNLOAD_DIR}" --no-same-owner
+        ExecStart=/usr/bin/rm "/opt/crictl-${CRICTL_VERSION}-linux-amd64.tar.gz"
+        [Install]
+        WantedBy=multi-user.target
+storage:
+  files:
+    - path: /opt/cni-plugins-linux-amd64-{{ .CNIVersion }}.tgz
+      filesystem: root
+      mode: 0644
+      contents:
+        remote:
+          url: https://github.com/containernetworking/plugins/releases/download/{{ .CNIVersion }}/cni-plugins-linux-amd64-{{ .CNIVersion }}.tgz
+          verification:
+            hash:
+              function: sha512
+              sum: a8fd7ffce7f4f1fb00ac2e4840b7961efc3ec844c5e99035c72baa95f038b316552ce71f8a17f82c04d614da490689953a7a80008d7b9be563187673813934b5
```

Comment:
Just an idea, maybe the checksum should also be a variable when the version is one

Reply:
I thought the same last week.
Having something like a `binary` struct 
```go
type Binary struct {
    Name string
    Version string
    SHA512 string
}
```

would be nice if tomorrow we want to test other versions